optimum/onnxruntime/modeling_ort.py

```diff
@@ -1059,18 +1059,37 @@ def forward(
         input_ids: Optional[Union[torch.Tensor, np.ndarray]] = None,
         attention_mask: Optional[Union[torch.Tensor, np.ndarray]] = None,
         token_type_ids: Optional[Union[torch.Tensor, np.ndarray]] = None,
-        **kwargs,
+        position_ids: Optional[Union[torch.Tensor, np.ndarray]] = None,
+        pixel_values: Optional[Union[torch.Tensor, np.ndarray]] = None,
+        input_features: Optional[Union[torch.Tensor, np.ndarray]] = None,
+        input_values: Optional[Union[torch.Tensor, np.ndarray]] = None,
     ):
-        use_torch = isinstance(input_ids, torch.Tensor)
+        # Determine the tensor type from any available tensor input
+        tensor_inputs = [
+            input_ids,
+            attention_mask,
+            token_type_ids,
+            position_ids,
+            pixel_values,
+            input_features,
+            input_values,
+        ]
+        first_tensor = next((t for t in tensor_inputs if t is not None), None)
+        use_torch = isinstance(first_tensor, torch.Tensor) if first_tensor is not None else False
```

Comment:
no need to check for first_tensor is None

```suggestion
        first_tensor = next(filter(None, tensor_inputs))
        use_torch = isinstance(first_tensor, torch.Tensor)
```